### PR TITLE
Problems view - add toggle to view problems in a tree/table

### PR DESCRIPTION
**Pull request by lszomoru**
This pull request adds a toggle to view the problems either as a tree (current behaviour) as well as in a table:

image

At a high level, I have introduce a new interface, `IProblemsWidget`, which is now implemented by both the `MarkersTree` and `MarkersTable` classes. I have also added a new setting, `problems.defaultViewMode`, to control the default view mode.

### CODE REVIEW

File: src/vs/workbench/contrib/markers/browser/markersModel.ts [hunk 1]
```diff
@@ -117,6 +118,19 @@ export class Marker {
 	}
 }
 
+export class MarkerTableItem extends Marker {
```
Comment (joaomoreno) [inline src/vs/workbench/contrib/markers/browser/markersModel.ts:121]: This feels weird. Wouldn't it be best if `MarkerTableItem` doesn't extend from `Marker`? Conceptually, `Marker` is a model object while `MarkerTableItem` seems like a view model object, no?

**Review by joaomoreno — DISMISSED** [review]
Works great!

File: src/vs/workbench/contrib/markers/browser/constants.ts [hunk 2]
```diff
@@ -23,6 +24,7 @@ export default {
 	MARKER_SHOW_QUICK_FIX: 'problems.action.showQuickFixes',
 	TOGGLE_MARKERS_VIEW_ACTION_ID: 'workbench.actions.view.toggleProblems',
 
+	MarkersViewModeContextKey: new RawContextKey<MarkersViewMode>('markersViewMode', MarkersViewMode.Tree),
```
[the remark this answers is not in the document]
Comment (lszomoru) [reply src/vs/workbench/contrib/markers/browser/constants.ts]: Thanks! Fixed.

File: src/vs/workbench/contrib/markers/browser/markersModel.ts (hunk 1 above)
Comment (lszomoru) [reply src/vs/workbench/contrib/markers/browser/markersModel.ts:121]: Happy to discuss this further to see if there is an implementation that feel more natural.

**Review by sandy081 — DISMISSED** [review]
Great job!!!

Approved - gave comments in person

File: src/vs/workbench/contrib/markers/browser/markersModel.ts (hunk 1 above)
Comment (lszomoru) [reply src/vs/workbench/contrib/markers/browser/markersModel.ts:121]: Closing this comment as @sandy081 and I have discussed this and agreed on a plan to address this.

**Review by joaomoreno — APPROVED** [review]
Amazing work.

**Pull request merged**